modules/polaris/src/test/ts/atomic/api/UrlTest.ts

```diff
@@ -0,0 +1,73 @@
+      'https://www.google.com.aa/test.htm?$-_.+!*\'()test,test;test:test@=&': 'google.com.aa',
+      'http://-.~_!$&\'()*+,;=:%40:80%2f::::::@example.com?-.~_!$&\'()*+,;=:%40:80%2f::::::@e#-.~_!$&\'()*+,;=:%40:80%2f::::::@e': 'e',
+      'http://xn--domain.com': 'xn--domain.com',
+      'www.google.ca/index.htm?id=/bla/bla': 'google.ca',
+      'https://www.amazon.com.au/gp/product/B0798R2WXG/ref=s9_acsd_top_hd_bw_b5QhTfX_c_x_w?pf_rd_m=ANEGB3WVEVKZB&pf_rd_s=merchandised-search-4&pf_rd_r=KF6SD7C0M69MKF2FR9CC&pf_rd_t=101&pf_rd_p=8ad3bdba-b846-5350-9c00-72c2cb7191dd&pf_rd_i=4975211051':
+        'amazon.com.au',
+      'https://www.birddoctor.net/refId,56511/refDownload.pml': 'birddoctor.net',
+      'https://www.example.com/:w:/s/b026324c6904b2a9cb4b88d6d61c81d1?q=abc123': 'example.com',
+      'https://website.com/test/!test': 'website.com',
+      'http://foo.com/blah_blah': 'foo.com',
+      'http://foo.com/blah_blah/': 'foo.com',
+      'http://foo.com/blah_blah_(wikipedia)': 'foo.com',
+      'http://foo.com/blah_blah_(wikipedia)_(again)': 'foo.com',
+      'http://www.example.com/wpstyle/?p=364': 'example.com',
+      'https://www.example.com/foo/?bar=baz&inga=42&quux': 'example.com',
+      'http://userid:password@example.com:8080': 'example.com',
+      'http://userid:password@example.com:8080/': 'example.com',
+      'http://userid@example.com': 'example.com',
+      'http://userid@example.com/': 'example.com',
+      'http://userid@example.com:8080': 'example.com',
+      'http://userid@example.com:8080/': 'example.com',
+      'http://userid:password@example.com': 'example.com',
+      'http://userid:password@example.com/': 'example.com',
+      'http://142.42.1.1/': '142.42.1.1',
+      'http://142.42.1.1:8080/': '142.42.1.1',
+      'http://foo.com/blah_(wikipedia)#cite-1': 'foo.com',
+      'http://foo.com/blah_(wikipedia)_blah#cite-1': 'foo.com',
+      'http://foo.com/(something)?after=parens': 'foo.com',
+      'http://code.google.com/events/#&product=browser': 'code.google.com',
+      'http://j.mp': 'j.mp',
+      'ftp://foo.bar/baz': 'foo.bar',
+      'http://foo.bar/?q=Test%20URL-encoded%20stuff': 'foo.bar',
+      'http://-.~_!$&\'()*+,;=:%40:80%2f::::::@example.com': 'example.com',
+      'http://1337.net': '1337.net',
+      'http://a.b-c.de': 'a.b-c.de',
+      'http://223.255.255.254': '223.255.255.254',
+      'h://foo.com': 'foo.com',
+      'h1://foo.com': 'foo.com',
+      'h1+://foo.com': 'foo.com',
+      'h1+.://foo.com': 'foo.com',
+      'h1+.-://foo.com': 'foo.com',
+      'p72internal://foo.com': 'foo.com',
+      'but15characters://foo.com': 'foo.com'
+    };
+
+    it('TINY-10350: Should correctly extract host name from urls', () => {
+      Obj.each(hostMap, (v, k) => {
+        const host = Url.extractHost(k).getOrDie();
+        Assert.eq(`expected host to be ${v} but was ${host}`, v, host);
+      });
+    });
```

Comment:
```suggestion
    Obj.each(hostMap, (v, k) => {
      it(`TINY-10350: Should correctly extract ${v} from ${k}`, () => {
        const host = Url.extractHost(k).getOrDie();
        Assert.eq(`expected host to be ${v} but was ${host}`, v, host);
      });
    });
```
Each test case should be its own test. This will allow us to tell what is failing.